packages/react/src/ProtectedRoute.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { FC, useEffect } from 'react';
+import { Route, RouteProps } from 'react-router-dom';
+
+import useLogto from './use-logto';
+
+const ProtectedRoute: FC<RouteProps> = ({ ...routeProperties }) => {
+  const { isAuthenticated, isLoading, isInitialized, loginWithRedirect } = useLogto();
+
+  useEffect(() => {
+    if (!isInitialized || isLoading || isAuthenticated) {
+      return;
+    }
+
+    void loginWithRedirect(window.location.href);
+  }, [isAuthenticated, isInitialized, loginWithRedirect]);
+
+  if (!isAuthenticated) {
+    return null;
```

Comment:
Do we intend to leave the developers to handler their own loading status apart from the `ProtectedRoute`?

Reply:
Yes, we just export `isInitialized` and `isLoading`